### PR TITLE
Refactor log analysis sns code to prepare for s3sns opstool

### PR DESCRIPTION
**Pull request by rleighton**
## Background

There will be a new opstool called `s3sns` that will allow replay of processed files through an sns topic.

This tool can replay files into log subscribers like Snowflake and can also be used to replay files through the rules engine.

This PR refactors some core code in preparation for the PR for the new opstool which will use some of this functionality.

## Changes

- The data catalog updater subscription had a filter added for Panther attributes
- Moved sns and s3 notification code into a package

## Testing

- mage test:ci
- mage deploy, partitions were created, logs processed

### CODE REVIEW

File: internal/log_analysis/notify/s3.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+import "github.com/aws/aws-lambda-go/events"
+
+// S3Notification is sent when new data is available in S3
+type S3Notification struct {
+	// https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html
+	Records []events.S3EventRecord
+}
+
+func NewS3ObjectPutNotification(bucket, key string, nbytes int) *S3Notification {
```
Comment (kostaspap) [inline internal/log_analysis/notify/s3.go:29]: Do we even use the `nbytes`? I think you can remove

File: internal/log_analysis/notify/sns.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+const (
+	logDataTypeAttributeName = "type"
+	logTypeAttributeName     = "id"
+)
+
+var (
+	messageAttributeDataType = "String"
+)
+
+func NewLogAnalysisSNSMessageAttributes(dataType, logType string) map[string]*sns.MessageAttributeValue {
```
Comment (kostaspap) [inline internal/log_analysis/notify/sns.go:34]: Better make it use `data DataType, logType string` -> https://github.com/panther-labs/panther/blob/20f774b7c3725ad2aaa00071e1d19ba185bbe824/api/lambda/core/log_analysis/log_processor/models/api.go#L22
DataType can be used to differentiate between the different types of events

File: internal/log_analysis/log_processor/destinations/s3.go [hunk 3]
```diff
@@ -288,18 +283,9 @@ func (destination *S3Destination) sendSNSNotification(key string, buffer *s3Even
...
-				StringValue: aws.String(models.LogData.String()),
-				DataType:    aws.String(messageAttributeDataType),
-			},
-			logTypeAttributeName: {
-				StringValue: aws.String(buffer.logType),
-				DataType:    aws.String(messageAttributeDataType),
-			},
-		},
+		TopicArn:          aws.String(destination.snsTopicArn),
+		Message:           aws.String(marshalledNotification),
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/destinations/s3.go:287]: nitpicking: Can just do `&destination.snsTopicArn`, `&marshalledNotification`

**Review by giorgosp — APPROVED** [review]
LGTM, just address Kostas' comments before merging.

File: internal/log_analysis/notify/s3.go (hunk 1 above)
Comment (rleighton) [reply internal/log_analysis/notify/s3.go:29]: Actually, this is super important. Snowflake won't load a 0 byte file. I'll put a comment to that effect.

Comment (rleighton) [reply internal/log_analysis/notify/s3.go:29]: Added a comment.

File: internal/log_analysis/notify/sns.go (hunk 2 above)
Comment (rleighton) [reply internal/log_analysis/notify/sns.go]: Done

File: internal/log_analysis/log_processor/destinations/s3.go (hunk 3 above)
Comment (rleighton) [reply internal/log_analysis/log_processor/destinations/s3.go]: Done

**Review by kostaspap — APPROVED** [review]

**Pull request merged**